squirtgun-platform-discord/src/main/java/net/lucypoulton/squirtgun/discord/standalone/StandaloneDiscordPlatform.java

```
@@ -0,0 +1,96 @@
+ * LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+ * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package net.lucypoulton.squirtgun.discord.standalone;
+
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.entities.Message;
+import net.lucypoulton.squirtgun.discord.DiscordPlatform;
+import net.lucypoulton.squirtgun.discord.DiscordUser;
+import net.lucypoulton.squirtgun.platform.audience.SquirtgunUser;
+import net.lucypoulton.squirtgun.platform.scheduler.TaskScheduler;
+import net.lucypoulton.squirtgun.plugin.SquirtgunPlugin;
+import org.jetbrains.annotations.Nullable;
+
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.Path;
+import java.util.UUID;
+import java.util.function.Predicate;
+import java.util.logging.Logger;
+
+/**
+ * A Platform implementation for standalone Discord bots.
+ */
+public class StandaloneDiscordPlatform extends DiscordPlatform {
+
+    private final TaskScheduler scheduler = new StandaloneTaskScheduler(this);
+    private final SquirtgunUser console = new StandaloneConsoleWrapper(this);
+
+
+    public StandaloneDiscordPlatform(JDA jda, String commandPrefix, Predicate<Message> commandPredicate) {
+        super(jda, commandPrefix, commandPredicate);
+        System.setProperty("java.util.logging.SimpleFormatter.format", "[%1$tF %1$tT] [%4$-7s] %5$s %n");
+    }
+
+    @Override
+    public DiscordUser getPlayer(UUID uuid) {
+        if (uuid.getMostSignificantBits() != 0L) return null;
+        return new StandaloneDiscordUser(jda().getUserById(uuid.getLeastSignificantBits()));
+    }
+
+    /**
+     * Throws an {@link IllegalStateException} - this method is <b>not supported standalone</b>
+     */
+    @Override
+    public @Nullable DiscordUser getPlayer(String name) {
+        throw new IllegalStateException("Standalone");
+    }
```

If "this method is **not supported**" it should either throw `UnsupportedOperationException` or simply return `null` given that it's annotated as nullable already.
If library users are using this in an abstracted enough way, they may be handling null cases but probably not unchecked exceptions.
Thoughts?

---

alternatively we could get the user by their username and discrim?

---

I like that idea, yeah, sounds totally valid

---

addressed in 7696cc1